spec/producers/payload_tracker_spec.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe PayloadTracker do
+  let(:request_id) { '1' }
```

Comment:
```suggestion
  let(:request_id) { Faker.random_int }
```

This could be something else than a hard-coded id, unless we really need to specify it in the unit test